native/screens/SignUp.jsx

```diff
@@ -1,180 +1,309 @@
+  const [newUser, setNewUser] = useState({
+    first_name: '',
+    last_name: '',
+    email: '',
+    password: '',
+    confirm_password: '',
+  });
+  // input references
+  const { 
+    firstNameInputRef, 
+    lastNameInputRef, 
+    emailInputRef, 
+    passwordInputRef, 
+    confirmPasswordInputRef } = createRef();
+
+  
+  const handleSignUp =  () => {
+    if (newUser.confirm_password !== newUser.password) {
+      setNewUser((prevState) => {
+        return {
+          ...prevState,
+          confirm_password: "",
+          password: "",
+        };
+      });
+      Alert.alert('Passwords do not match. Please try again.')
+    } else if (emailError) {
+      Alert.alert(`Account with email ${newUser.email} already exists. Please use a different email or sign in to your account.`)
+      setNewUser((prevState) => {
+        return {
+          ...prevState,
+          confirm_password: "",
+          password: "",
+        };
+      });
+      setEmailError(null);
+    } else {
+      dispatch(signUpUser(newUser));
+      setSuccessModalVisible(true);
+      setNewUser({
+        first_name: '',
+        last_name: '',
+        email: '',
+        password: '',
+        confirm_password: '',
+      })
+    }
+  };
 
-//   const handleSignUp = async (event) => {
-//     event.preventDefault();
-//     if (newUser.confirm_password !== newUser.password) {
-//       setNewUser((prevState) => {
-//         return {
-//           ...prevState,
-//           confirm_password: "",
-//           password: "",
-//         };
-//       });
-//       setModalError("Passwords do not match. Please try again.");
-//       setShowModal(true);
-//     } else if (emailError) {
-//       setModalError(
-//         "An account with this email already exists. Please try another email or Sign in."
-//       );
-//       setShowModal(true);
-//       setNewUser((prevState) => {
-//         return {
-//           ...prevState,
-//           email: "",
-//           confirm_password: "",
-//           password: "",
-//         };
-//       });
-//       setEmailError(null);
-//     } else {
-//       dispatch(signUpUser(newUser));
-//       setShowSuccessModal(true);
-//     }
-//   };
+  // Check if email is already in use
+  const handleEmailCheck = async () => {
+    const emailReq = {email: newUser.email}
+    const res = await checkEmailAuth(emailReq);
+    if (res) {
+      setEmailError(res);
+    }
+  };
 
-//   const handleEmailCheck = async (e) => {
-//     const emailReq = { email: e.target.value };
-//     const res = await checkEmailAuth(emailReq);
-//     if (res) {
-//       setEmailError(res);
-//     }
-//   };
+  const handleReroute = async (screen) => {
+    let id = null;
+    if (screen === 'EditProfile') {
+      let resp = await verify();
+      id = resp._id;
+    }
+    setSuccessModalVisible(false);
+    // need to test this params funcionality on edit profile screen
+    navigation.navigate(screen, {
+      id: id,
```

Comment:
although there is no other ID on this file, it may not be obvious to the dev that accesses this `id` param on the next screen, so let's be semantic about passing in the userID to the next screen.